LoRaEngine/test/LoRaWanNetworkServer.Test/MessageProcessorJoinTest.cs

```diff
@@ -418,5 +418,409 @@ public async Task When_Api_Returns_DevAlreadyUsed_Should_Return_Null()
             this.LoRaDeviceApi.VerifyAll();
             this.LoRaDeviceClient.VerifyAll();
         }
+
+        [Theory]
+        [InlineData(1, 1)]
+        [InlineData(3, 4)]
+        [InlineData(2, 0)]
+        public async Task When_Getting_DLSettings_From_Twin_Returns_JoinAccept_With_Correct_Settings(int rx1DROffset, int rx2datarate)
+        {
+            string deviceGatewayID = ServerGatewayID;
+            var simulatedDevice = new SimulatedDevice(TestDeviceInfo.CreateOTAADevice(1, gatewayID: deviceGatewayID));
+            var joinRequest = simulatedDevice.CreateJoinRequest();
+
+            // Create Rxpk
+            var rxpk = joinRequest.SerializeUplink(simulatedDevice.LoRaDevice.AppKey).Rxpk[0];
+
+            var devNonce = ConversionHelper.ByteArrayToString(joinRequest.DevNonce);
+            var devAddr = string.Empty;
+            var devEUI = simulatedDevice.LoRaDevice.DeviceID;
+            var appEUI = simulatedDevice.LoRaDevice.AppEUI;
+
+            // Device twin will be queried
+            var twin = new Twin();
```

Comment:
fyi: there is an extension method to create OTAA/ABP twins, so you don't have to hardcode all values.

Reply:
TBH, I just inspired myself from the test When_Getting_Device_Information_From_Twin_Returns_JoinAccept but this is a good point. I guess as we are in Strict mode I would still need to setup gettwinasync right? The method you mention is simulatedDevice.CreateOTAATwin correct?

Reply:
Correct. Just for the next time you write a test ;)

Reply:
I can change them, it's no problems